File: HW/HW8/longestPalindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        char_count = {}

        for char in s:
            char_count[char] = char_count.get(char, 0) + 1

        palindrome_length = 0
        has_odd_count = False

        for count in char_count.values():
            palindrome_length += count // 2 * 2
            if count % 2 == 1:
                has_odd_count = True

        return palindrome_length + 1 if has_odd_count else palindrome_length

solution = Solution()
print(solution.longestPalindrome("abccccdd"))  

File: HW/HW8/longestPalindrome.java
public class Solution {
    public int longestPalindrome(String s) {
        int[] charCount = new int[128];
        int palindromeLength = 0;

        for (char c : s.toCharArray()) {
            charCount[c]++;
        }

        for (int count : charCount) {
            palindromeLength += count / 2 * 2;
            if (count % 2 == 1 && palindromeLength % 2 == 0) {
                palindromeLength++;
            }
        }

        return palindromeLength;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.longestPalindrome("abccccdd"));
    }
}
